### ISSUE TITLE
Docker problem: libxml extension failing

### ISSUE THREAD

**Issue by bwalsh** [ga4gh/server#484]
Attempted to build docker image on macos

```
    copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/readme.txt -> build/lib.linux-x86_64-2.7/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
    running build_ext
    building 'lxml.etree' extension
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/src
    creating build/temp.linux-x86_64-2.7/src/lxml
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/tmp/pip-build-cWMb1G/lxml/src/lxml/includes -I/usr/local/python/include/python2.7 -c src/lxml/lxml.etree.c -o build/temp.linux-x86_64-2.7/src/lxml/lxml.etree.o -w
    In file included from src/lxml/lxml.etree.c:239:0:
    /tmp/pip-build-cWMb1G/lxml/src/lxml/includes/etree_defs.h:14:31: fatal error: libxml/xmlversion.h: No such file or directory
     #include "libxml/xmlversion.h"
                                   ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/local/python/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-build-cWMb1G/lxml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-QxYF6g-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-cWMb1G/lxml
The command '/bin/sh -c mod_wsgi-docker-build' returned a non-zero code: 1
```

**Comment by jeromekelleher** [comment]
Sorry, I can't help here @bwalsh, I don't have access to a mac. Any apple fans able to reproduce this?

**Comment by bwalsh** [comment]
Hi.  Thanks for the follow-up.  I'm not sure it is a macos problem.

I experimented a bit more and found that if I created a image based on this Dockerfile  ...

```
FROM grahamdumpleton/mod-wsgi-docker:python-2.7-onbuild
RUN apt-get update
RUN apt-get -y install python2.7-dev libxml2-dev libxslt1-dev
```

Then built server image based on that image, the build was successful.

I'm not sure what is the best way to propose a fix; since `.whiskey/action_hooks/pre-build` executes directly after the `FROM`, it is not as simple as adding the apt-get to the Dockerfile.  I'm afraid I'm not 100% up-to-speed on action hooks.

**Comment by jeromekelleher** [comment]
Thanks for updating on this one @bwalsh. I'm afraid I'm almost completely clueless when it comes to Docker, so I don't really know what the appropriate action is here. @afirth, can you give us some pointers on what we should do here please?

**Comment by afirth** [comment]
I get a different error on dockerhub, which seems to have broken with #447 installing cherrypy, I imagine it will be broken for some local builds too. It shouldn't be mac specific (I assume you are building on boot2docker, which is a vm build from the  tiny core linux distro). I'll dig into it more when I get a chance.

**Comment by dcolligan** [comment]
Fixed in #561
